File: push-client-java/src/main/java/wang/ismy/push/client/Connector.java
package wang.ismy.push.client;

import java.io.IOException;

/**
 * 服务端Connector在客户端的表现
 * @author MY
 * @date 2020/9/9 20:32
 */
public class Connector {

    private String lookupAddress;
    private String host;
    private int port;
    private HttpTemplate httpTemplate = new HttpTemplate();

    public Connector(String lookupAddress, HttpTemplate httpTemplate) {
        this(lookupAddress);
        this.httpTemplate = httpTemplate;
    }

    public Connector(String lookupAddress) {
        this.lookupAddress = lookupAddress;
    }

    /**
     * 请求分配一个connector
     * @return 返回是否分配成功
     * @throws IOException
     * @throws InterruptedException
     */
    public boolean lookupConnector() throws IOException, InterruptedException {
        String response = httpTemplate.get(lookupAddress);
        String[] splitResult = response.split(":");
        if (splitResult.length != 2) {
            return false;
        }

        host = splitResult[0];
        port = Integer.parseInt(splitResult[1]);
        return true;
    }

    public boolean isAvailable(){
        return host != null && !"".equals(host) && port > 0;
    }

    public String getHost() {
        return host;
    }

    public int getPort() {
        return port;
    }
}

File: push-client-python/Connector.py
from http_template import *
import requests

class Connector:
    def __init__(self, lookup_address):
        self.lookup_address = lookup_address
        self.host = ''
        self.port = 0

    def lookup_connector(self):
        response = requests.get(self.lookup_address).text
        splitResult = response.split(':')
        if len(splitResult) != 2:
            raise IOError('获取 connector 失败')
        self.host = splitResult[0]
        self.port = int(splitResult[1])

    def is_available(self):
        return self.host != '' and self.port > 0


connector = Connector('http://192.168.1.100:30001')
print (connector.is_available())
connector.lookup_connector()
print (connector.is_available())
print(connector.host)
print(connector.port)